File: Sources/App/DTOs/PlaylistDTO.swift
//
//  PlaylistDTO.swift
//  FinchServer
//
//  Created by Stefan Klein Nulent on 16/02/2025.
//

import Vapor

struct PlaylistDTO: Content {
    
    // MARK: - Types
    
    enum CodingKeys: String, CodingKey {
        case id
        case name
        case description
        case image
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
    
    
    
    // MARK: - Properties
    
    let id: Int?
    let name: String
    let description: String?
    let image: String?
    let createdAt: Date?
    let updatedAt: Date?
    
    
    
    // MARK: - Construction
    
    init(_ playlist: Playlist) {
        id = playlist.id
        name = playlist.name
        description = playlist.description
        image = playlist.image
        createdAt = playlist.createdAt
        updatedAt = playlist.updatedAt
    }
}

File: FinchServer/Controllers/DTO/PlaylistDto.cs
using FinchServer.Database;

namespace FinchServer.Controllers.DTO;

public class PlaylistDto(Playlist playlist) {
    
    // - Properties
    
    public int Id { get; set; } = playlist.Id;
    public string Name { get; set; } = playlist.Name;
    public string? Description { get; set; } = playlist.Description;
    public bool IsArtworkAvailable { get; set; } = playlist.ImageFileName != null;
    public DateTimeOffset CreatedAt { get; set; } = playlist.CreatedAt.AddTicks( - (playlist.CreatedAt.Ticks % TimeSpan.TicksPerSecond));
    public DateTimeOffset UpdatedAt { get; set; } = playlist.UpdatedAt.AddTicks( - (playlist.UpdatedAt.Ticks % TimeSpan.TicksPerSecond));
}